subprojects/instant-execution/src/main/kotlin/org/gradle/instantexecution/InstantExecutionState.kt

```diff
@@ -217,8 +227,8 @@ class InstantExecutionState(
 
 
 internal
-fun fillTheGapsOf(projects: Collection<Project>): List<Project> {
-    val projectsWithoutGaps = ArrayList<Project>(projects.size)
+fun fillTheGapsOf(projects: Sequence<Project>): List<Project> {
+    val projectsWithoutGaps = ArrayList<Project>()
```

Comment:
❌  How about using a `LinkedHashSet<Project>` here? The return type can be changed to `Collection<Project>`.